hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -1651,7 +1652,7 @@ private void putAllInternal(@Nonnull Map<? extends K, ? extends V> map,
             ClientMessage request = MapPutAllCodec.encodeRequest(name, entry.getValue(), triggerMapLoader);
             new ClientInvocation(getClient(), request, getName(), partitionId)
                     .invoke()
-                    .whenCompleteAsync(callback);
+                    .whenCompleteAsync(callback, ConcurrencyUtil.getDefaultAsyncExecutor());
```

Comment:
this callback is also relevant for near cache upkeep, maybe `client#getTaskScheduler()`?